src/core/thread/mle.cpp

```diff
@@ -2508,8 +2494,6 @@ otError Mle::SendAnnounce(uint8_t aChannel, bool aOrphanAnnounce, const Ip6::Add
     {
         message->Free();
```

Comment:
log failure? 

Would it be better if we use the same statement for both successful log and failure log, like:
```suggestion

if (error == OT_ERROR_NONE) {
  otLogInfoMle("Send Announce on channel %d: %s", aChannel, otThreadErrorToString(error));
} else {
  otLogWarnMle("Send Announce on channel %d: %s", aChannel, otThreadErrorToString(error));
}

        message->Free();
```
We can define a new MACRO for this purpose.

Reply:
We can consider this, but probably better for a separate PR that covers all instances that are not within scope of this PR.

Reply:
Agree